src/history/history-entry.enity.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import {
+  Column,
+  Entity,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { User } from '../users/user.entity';
+import { Note } from '../notes/note.entity';
+
+@Entity()
+export class HistoryEntry {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @ManyToOne((_) => User, (user) => user.historyEntries, {
+    onDelete: 'CASCADE',
+  })
+  user: User;
+
+  @ManyToOne((_) => Note, (note) => note.historyEntries, {
+    onDelete: 'CASCADE',
+  })
+  note: Note;
+
+  @Column()
+  pinStatus: boolean;
+
+  @UpdateDateColumn()
+  updatedAt: Date;
+
+  public static create(user: User, note?: Note): HistoryEntry {
+    const newHistoryEntry = new HistoryEntry();
+    newHistoryEntry.user = user;
+    if (note) {
```

Comment:
A HistoryEntry without a note does not make much sense to me. Why do we make it optional?

(The column is also missing a nullable annotation, *if* `note` needs to be optional)

Reply:
No this quite right. This is needed because of the way TypeORM works.
The create function without a note is used, when a note is created, in that case TypeORM adds the appropriate noteId to the database.

Reply:
I added a comment